laiaaa0/assignment2/BinaryTree.cpp

```diff
@@ -0,0 +1,164 @@
+#include "BinaryTree.h"
+
+//BinaryTree::BinaryTreeInorder(){}
+//BinaryTree::BinaryTreePostorder();
+template <class T>
+BinaryTree<T>::BinaryTree(){
+  this->left_subtree_ = NULL;
+  this->right_subtree_ = NULL;
+  this->parent_ = NULL;
+}
+template <class T>
+BinaryTree<T>::BinaryTree(const T & value){
+  this->root_value_ = value;
+  this->left_subtree_ = NULL;
+  this->right_subtree_ = NULL;
+  this->parent_ = NULL;
+}
+template <class T>
+BinaryTree<T>::BinaryTree(const T & value, BinaryTree<T> * left, BinaryTree<T> * right){
+  this->root_value_ = value;
+  this->left_subtree_ = left;
+  this->right_subtree_ = right;
+  this->left_subtree_->parent_ = this;
+  this->right_subtree_->parent_ = this;
+}/*
+template <class T>
+BinaryTree<T>::BinaryTree(std::queue<T> &values, InputMethod method){
+
+  switch (method) {
+    case PREORDER:
+    {
+      if (!values.empty()){
+
+        this->root_value_ = values.front();
+        values.pop();
+        BinaryTree<T> *left_tree = new BinaryTree<T>(values,method);
+        this->left_subtree_ = left_tree;
+        BinaryTree<T> * right_tree = new BinaryTree<T>(values,method);
+        this->right_subtree_ = right_tree;
+      }
+      break;
+    }
+
+    case INORDER:
+      break;
+    default:
+      break;
+  }
+
+}*/
+template <class T>
+const T & BinaryTree<T>::GetRootValue(){
+  return this->root_value_;
+}
+
+template <class T>
+BinaryTree<T> * BinaryTree<T>::GetRightSubtree(){
+  return this->right_subtree_;
+}
+
+template <class T>
+BinaryTree<T> * BinaryTree<T>::GetLeftSubtree(){
+  return this->left_subtree_;
+}
+
+
+template <class T>
+bool BinaryTree<T>::IsValueInTree (const T & value, BinaryTree<T>* &root_of_value){
```

Comment:
`*&` is a bit confusing :)
Optionally, you can use a `**` parameter or return a pointer (and nullptr in case of not found).
Also, the name of he function should indicate that you are actually finding and returning the node. Something like `findValue` or similar, maybe?